tests/sentry/hybrid_cloud/test_organizationmapping.py

```diff
@@ -1,93 +1,48 @@
-        )
-        assert rpc_org_mapping.verified is org_mapping.verified is False
-        assert rpc_org_mapping.slug == org_mapping.slug == fields["slug"]
-        assert rpc_org_mapping.region_name == org_mapping.region_name == fields["region_name"]
-        assert rpc_org_mapping.date_created == org_mapping.date_created
-        assert rpc_org_mapping.name == org_mapping.name == fields["name"]
-
-    def test_idempotency_key(self):
-        self.organization = Factories.create_organization(no_mapping=True)
-        data = {
-            "slug": self.organization.slug,
-            "name": "test name",
-            "region_name": "us",
-            "idempotency_key": "test",
-        }
-        self.create_organization_mapping(self.organization, **data)
-        next_organization_id = 7654321
-        rpc_org_mapping = organization_mapping_service.create(
-            **{**data, "organization_id": next_organization_id}
-        )
+    def test_create_on_organization_save(self):
+        with exempt_from_silo_limits():
+            self.organization = Organization(
+                name="test name",
+            )
+            self.organization.save()
 
-        assert not OrganizationMapping.objects.filter(organization_id=self.organization.id).exists()
-        org_mapping = OrganizationMapping.objects.filter(organization_id=next_organization_id)
-        assert org_mapping
-        org_mapping = org_mapping.first()
-        assert org_mapping.idempotency_key == data["idempotency_key"]
-
-        assert rpc_org_mapping.organization_id == next_organization_id
-        assert rpc_org_mapping.region_name == "us"
-        assert rpc_org_mapping.name == data["name"]
-
-    def test_duplicate_slug(self):
-        self.organization = Factories.create_organization(no_mapping=True)
-        data = {
-            "slug": self.organization.slug,
-            "name": "test name",
-            "region_name": "us",
-            "idempotency_key": "test",
-        }
-        self.create_organization_mapping(self.organization, **data)
+        # Validate that organization mapping has not been created
+        with pytest.raises(OrganizationMapping.DoesNotExist):
+            OrganizationMapping.objects.get(organization_id=self.organization.id)
 
-        with pytest.raises(IntegrityError):
-            organization_mapping_service.create(
-                **{
-                    **data,
-                    "organization_id": 7654321,
-                    "region_name": "de",
-                    "idempotency_key": "test2",
-                }
-            )
+        # Drain outbox to ensure mapping is created
+        with outbox_runner():
+            pass
+
+        org_mapping = OrganizationMapping.objects.get(organization_id=self.organization.id)
+        assert org_mapping.idempotency_key == ""
+        assert self.organization.id == self.organization.id
```

Comment:
```suggestion
        assert self.organization.id == org_mapping.organization_id
```

The previous assertion wasn't doing anything.

Reply:
🙈 Oops, thank you